main.go

```diff
@@ -961,9 +970,17 @@ func (task *TaskRun) ExecuteCommand(index int) *CommandExecutionError {
 
 	switch {
 	case result.Failed():
-		return &CommandExecutionError{
-			Cause:      result.FailureCause(),
-			TaskStatus: failed,
+		if task.RetryExitCode(int64(result.ExitCode())) {
+			return &CommandExecutionError{
+				Cause:      fmt.Errorf("Intermittent task will be rerun. Exit code was in onExitStatusList"),
```

Comment:
Let's include the code and not mention rerun, as that depends on the Queue (whose behaviour may change to e.g. creating a retry rather than rerun in future).
```
Cause: fmt.Errorf("Task appears to have failed intermittently - exit code %v found in task payload.onExitStatus list", result.ExitCode()),
```